addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3980,6 +3994,89 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    /**
+     * Places the cursor in a safe place (not the editable root).
+     * Inserts an empty paragraph when there's no other way to insert text
+     * before/after a block.
+     * If selection was result of an arrow key press, no new 'P' is inserted.
+     *
+     * @param {Selection} selection
+     * @param {String} currentKeyPress
+     */
+    _fixSelectionOnEditableRoot(selection, currentKeyPress) {
+        if (!selection || selection.anchorNode !== this.editable) return;
+        const nodesToSkip = ['#text', "#comment"];
+        if (currentKeyPress && currentKeyPress.startsWith("Arrow")) {
+            nodesToSkip.push('HR');
+        }
+        let nodeAfterCursor = this.editable.childNodes[selection.anchorOffset];
+        let nodeBeforeCursor = nodeAfterCursor && nodeAfterCursor.previousSibling;
+        while (nodeAfterCursor && nodesToSkip.includes(nodeAfterCursor.nodeName)) {
+            nodeAfterCursor = nodeAfterCursor.nextSibling;
+        }
+        while (nodeBeforeCursor && nodesToSkip.includes(nodeBeforeCursor.nodeName)) {
+            nodeBeforeCursor = nodeBeforeCursor.previousSibling;
+        }
```

Comment:
You're right.
`nodeBeforeCursor` is simply `nodeAfterCursor.previousElementSibling` (no loop).
Skipping void elements is only necessary when handling the arrow key presses, so I moved the logic to inside those blocks.
And I used `!node.chilNodes.length`  as a test for "void" element. It's not exactely the same thing, as "having no child nodes" is different than" cannot have child nodes" (definition of void element)... but I think that's true in OdooEditor (There's always at least some `<br>` or ZWS in any of our non-void elements... right??) ?

Alternatively, we have an util called `isVoidElement`. It would handle the 'HR' case, but not the others in this list: https://developer.mozilla.org/en-US/docs/Glossary/Void_element.

Reply:
A possible test for void element would be `!element.outerHTML.includes('</')`

Reply:
The `isVoidElement` util can be fixed to include the other elements. It used to be an obscure util from the GA which only matched like BR I think, we added HR to it to fix a bug and there is another task to add IMG to it too. Theoretically it should have _all_ void elements in it. My point is: use this util and feel free to adapt it if you think it does not handle all the relevant cases.

Reply:
Adding `|| node.tagName === 'BR'` to ` isVoidElement` simply breaks 63 tests... :cry: 
`isVoidElement` is used by other pieces of code like:
- @ sanitize:
```
           // Ensure elements which should not contain any content are tagged
            // contenteditable=false to avoid any hiccup.
            if (
                isVoidElement(node) &&
                node.getAttribute('contenteditable') !== 'false'
            ) {
                node.setAttribute('contenteditable', 'false');
            }
```
- @ OdooEditor._activateContentEditable:
```
        const editableAreas = this.options.getContentEditableAreas(this).filter(node => !isVoidElement(node));
        for (const node of editableAreas) {
            if (!node.isContentEditable) {
                node.setAttribute('contenteditable', true);
            }
        }
```
We (normally) would not expect to find a BR at the editable root, so I decided to not touch this function, as it covers our 'HR' case as it is.
Agreed?

Reply:
@dmo-odoo Right I see... So we have 2 concepts: (1) the element is void, and (2) the element is _not allowed_ to have content. Void elements (like `br` or `img`) are not allowed to have content but that's by original design so the browser won't allow it anyway and we don't need to worry about it. But there are other elements that we have to manage ourselves (like `i.fa`).
![image](https://user-images.githubusercontent.com/24205914/233560752-784db893-3526-420f-b37b-33319f6ddfbc.png)
Is that correct so far?
If so, we could keep `isVoidElement` and add something to the effect of:
```js
function isFakeVoidElement(node) {
    return node.classList.contains('fa') || node.classList.contains('o_fake_void');
function isNotAllowedContent(node) {
    return isVoidElement(node) || isFakeVoidElement(node);
```